### PR TITLE
[terra-functional-testing] Allow consumers to set maxInstances while executing WDIO.

### PR DESCRIPTION
**Pull request by saket2403**
### Summary

**What was changed:**
Added feature to allow consumers to set maxInstances while running wdio.

**Why it was changed:**

It was required by a consuming team.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9130 

---

Thank you for contributing to Terra.
@cerner/terra

Tested with terra-core: 10 instances
image

### CODE REVIEW

**Comment by supreethmr** [comment]
@saket2403 Can we get this fix validated with terra-core / framework and orion-example-component-JS

File: packages/terra-functional-testing/src/config/utils/getMaxInstances.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = () => {
+  const args = process.argv.slice(2);
+  const maxInstancesIndex = args.indexOf('--maxInstances');
+  const maxInstances = maxInstancesIndex !== -1 ? parseInt(args[maxInstancesIndex + 1], 10) : 1;
```
Comment (supreethmr) [inline packages/terra-functional-testing/src/config/utils/getMaxInstances.js:4]: why args[maxInstancesIndex + 1] index +1 here ..??

File: packages/terra-functional-testing/src/config/utils/getMaxInstances.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = () => {
+  const args = process.argv.slice(2);
```
Comment (supreethmr) [inline packages/terra-functional-testing/src/config/utils/getMaxInstances.js]: why do we need to slice 2 here ..?

File: packages/terra-functional-testing/src/config/utils/getMaxInstances.js (hunk 1 above)
Comment (saket2403) [reply packages/terra-functional-testing/src/config/utils/getMaxInstances.js:4]: Beacuse the argv array contains `--maxInstances` as an item whose index I am retrieving in the line before this followed by the value so +1.

**Comment by sugan2416** [comment]
> @saket2403 Can we get this fix validated with terra-core / framework and orion-example-component-JS

Tested with terra-core: 10 instances
image

File: packages/terra-functional-testing/src/config/utils/getMaxInstances.js (hunk 2 above)
Comment (saket2403) [reply packages/terra-functional-testing/src/config/utils/getMaxInstances.js]: Not required removed [bb1cedd](https://github.com/cerner/terra-toolkit/pull/837/commits/bb1cedd1374af23a3587607da5145008cb3d688b)

File: packages/terra-functional-testing/src/config/utils/getMaxInstances.js (hunk 1 above)
Comment (supreethmr) [reply packages/terra-functional-testing/src/config/utils/getMaxInstances.js:4]: can we access the value with name args['--maxInstances'];

**Review by supreethmr — APPROVED** [review]

**Review by sugan2416 — APPROVED** [review]

**Pull request merged**